module_clean_urls.lua

```diff
@@ -0,0 +1,737 @@
+    ["x.com"] = "fixvx.com",
+}
+
+Module.UsersHanging = {}
+
+Module.UniversalRules = {}
+Module.HostRules = {}
+Module.RulesByHost = {}
+Module.GuildUniversalRules = {}
+Module.GuildHostRules = {}
+Module.GuildRulesByHost = {}
+
+function Module:CreateRules()
+    -- Can be extended with a config option in the future
+    local rules = defaultRules
+
+
+    for _, rule in ipairs(rules) do
+        ---@diagnostic disable-next-line: undefined-field
+        local splitRule = rule:split("@")
+        local pattern = "^" .. escapeRegex(splitRule[1]):gsub("%%%*", ".-") .. "$"
+
+        if not splitRule[2] then
+            table.insert(self.UniversalRules, pattern)
+        else
+            local hostPattern = "^(w*%.?)" .. "(" .. escapeRegex(splitRule[2])
+                :gsub("\\%.", "\\.")
+                :gsub("^%%%*%.", "(.-%.)?")
+                :gsub("%%%*", ".-") .. ")" .. "$"
+
+            local hostPatternIndex = hostPattern
+
+            self.HostRules[hostPatternIndex] = hostPattern
+
+            if not self.RulesByHost[hostPatternIndex] then
+                self.RulesByHost[hostPatternIndex] = {}
+            end
+
+            table.insert(self.RulesByHost[hostPatternIndex], pattern)
+        end
+    end
+
+    return self.UniversalRules, self.HostRules, self.RulesByHost
+end
+
+function Module:CreateGuildRules(config, guildId)
+    local rules = config.Rules
+
+    local universalRules = self.GuildUniversalRules[guildId] or {}
+    local hostRules = self.GuildHostRules[guildId] or {}
+    local rulesByHost = self.GuildRulesByHost[guildId] or {}
+
+    for _, rule in ipairs(rules) do
+        ---@diagnostic disable-next-line: undefined-field
+        local splitRule = rule:split("@")
+        local pattern = "^" .. escapeRegex(splitRule[1]):gsub("%%%*", ".-") .. "$"
+
+        if not splitRule[2] then
+            table.insert(universalRules, pattern)
+        else
+            local hostPattern = "^(w*%.?)" .. escapeRegex(splitRule[2])
+                :gsub("\\%.", "\\.")
+                :gsub("^%%%*%.", "(.-%.)?")
+                :gsub("%%%*", ".-") .. "$"
+
+            local hostPatternIndex = hostPattern
+
+            hostRules[hostPatternIndex] = hostPattern
+
+            if not rulesByHost[hostPatternIndex] then
+                rulesByHost[hostPatternIndex] = {}
+            end
+
+            table.insert(rulesByHost[hostPatternIndex], pattern)
+        end
+    end
+
+    self.GuildUniversalRules[guildId] = universalRules
+    self.GuildHostRules[guildId] = hostRules
+    self.GuildRulesByHost[guildId] = rulesByHost
+end
+
+function Module:ClearGuildRules(config, guildId)
+    self.GuildUniversalRules[guildId] = nil
+    self.GuildHostRules[guildId] = nil
+    self.GuildRulesByHost[guildId] = nil
+end
+
+local function removeParam(rule, param, queryParams)
+    if param == rule or string.match(param, rule) then
+        queryParams[param] = nil
+    end
+end
+
+local function request(method, url)
```

Comment:
what's the point vs coro http.request?

Reply:
Accidental push while testing some things, will remove it